py/dml/serialize.py

```diff
@@ -118,28 +118,27 @@ def construct_assign_apply(funname, intype):
                                  [current_expr], attr_value_t)
         return ctree.mkAssignStatement(current_site, target_expr,
                                        ctree.ExpressionInitializer(apply_expr))
-    if isinstance(real_type, TInt):
-        if real_type.signed:
-            return construct_assign_apply("SIM_make_attr_int64", real_type)
-        else:
-            return construct_assign_apply("SIM_make_attr_uint64", real_type)
-    if isinstance(real_type, TEndianInt):
```

Comment:
Turns out `deserialize` already _does_ do this properly! It's just `serialize` that is bad.